jaeger-core/src/test/java/io/jaegertracing/internal/propagation/TraceContextCodecTest.java

```diff
@@ -160,10 +164,31 @@ public void testEmptyTraceStateNotPropagated() {
     assertEquals(EXAMPLE_TRACE_PARENT, injectCarrier.get(TRACE_PARENT));
   }
 
+  @Test
+  public void testDebugIdWithoutTraceHeader() {
+    Map<String, String> extractCarrier = new HashMap<>();
+    TextMapAdapter textMap = new TextMapAdapter(extractCarrier);
+    textMap.put(Constants.DEBUG_ID_HEADER_KEY, EXAMPLE_DEBUG_ID);
+    JaegerSpanContext spanContext = traceContextCodec.extract(textMap);
+    assertTrue(spanContext.isDebug());
+  }
+
+  @Test
+  public void testDebugIdWithTraceHeader() {
+    Map<String, String> extractCarrier = new HashMap<>();
+    TextMapAdapter textMap = new TextMapAdapter(extractCarrier);
+    textMap.put(TRACE_PARENT, EXAMPLE_TRACE_PARENT);
+    textMap.put(Constants.DEBUG_ID_HEADER_KEY, EXAMPLE_DEBUG_ID);
+    JaegerSpanContext spanContext = traceContextCodec.extract(textMap);
+    assertTrue(spanContext.isDebug());
+  }
+
   private void verifyWarningPresent() {
     ArgumentCaptor<byte[]> captor = ArgumentCaptor.forClass(byte[].class);
     try {
-      verify(System.out).write(captor.capture());
+      verify(System.out, atLeast(0)).write(captor.capture(), anyInt(), anyInt());
+      // gradle will call write(byte[]) instead of write(byte[],off,len)
```

Comment:
What does "gradle" have to do with unit test?

Reply:
?

Reply:
gradle output log via `write(byte[])` and eclipse output log via `write(byte[],int,int)` , before this change, tests failed with eclipse junit runner. I know this is fragile, but I cannot find a better way to assert log message.